Gordon360/Services/ProfileService.cs

```diff
@@ -301,6 +301,48 @@ public async Task UpdateMobilePrivacyAsync(string username, string value)
             _context.SaveChanges();
         }
 
+        /// <summary>
+        /// update planned graduation
+        /// </summary>
+        /// <param name="username">The username for the user whose planned graduation year is to be updated</param>
+        /// <param name="newPlannedGraduationYear">The new year to update the user's planned graduation year to</param>
+        public async Task<StudentProfileViewModel> UpdatePlannedGraduationYearAsync(string username, string newPlannedGraduationYear)
+        {
+            var profile = GetStudentProfileByUsername(username);
+            if (profile == null)
+            {
+                throw new ResourceNotFoundException() { ExceptionMessage = "The account was not found" };
+            }
+
+            // Update value in cached data
+
+            var student = _context.CUSTOM_PROFILE.FirstOrDefault(x => x.username == username);
```

Comment:
```suggestion
            var student = _context.CUSTOM_PROFILE.Find(username);
```
If I'm not mistaken `custom_profile` has a primary key of ad_username